hadoop-ozone/dist/src/main/compose/upgrade/upgrades/non-rolling-upgrade/1.2.1-1.3.0/callback.sh

```diff
@@ -70,7 +70,7 @@ with_old_version_downgraded() {
 }
 
 with_new_version_finalized() {
-  _check_hdds_mlvs 4
+  _check_hdds_mlvs 5
```

Comment:
Since 1.3.0 is released, I think we should keep this at 4, and introduce a new upgrade subdir for 1.3.0-1.4.0.

Reply:
In offline discussion with @dombizita and @fapifta we agreed that the 1.4.0 upgrade test should be added separately, before this PR.  Thanks @errose28 for the info on your work related to that.

Also, ideally this PR should be split into two:

* adding the framework to handle new ports, and applying it to RATIS_DATASTREAM
* adding the new HTTP ports

each with a new layout version, and bumping expected version after finalization.

Reply:
okay, thanks @errose28 @adoroszlai @fapifta, I converted this PR to draft and I'll come back to it once the other two are merged.

Reply:
See #4456 for HDDS-6633 PR.